control-base/src/main/java/fi/nls/oskari/control/feature/FeatureWFSTRequestBuilder.java

```diff
@@ -0,0 +1,113 @@
+import javax.xml.stream.XMLStreamWriter;
+import java.io.*;
+import java.util.Map;
+
+public class FeatureWFSTRequestBuilder extends WFSTRequestBuilder {
+    private final static Logger LOG = LogFactory.getLogger(FeatureWFSTRequestBuilder.class);
+
+    public static void updateFeature(OutputStream out, Feature feature)
+            throws XMLStreamException {
+        XMLStreamWriter xsw = XOF.createXMLStreamWriter(out);
+        writeStartTransaction(xsw, "1.1.0");
+
+        xsw.writeStartElement(WFS, "Update");
+        xsw.writeAttribute("typeName", feature.getLayerName());
+
+        for (Map.Entry<String, String> property : feature.getProperties().entrySet()) {
+            xsw.writeStartElement(WFS, "Property");
+
+            xsw.writeStartElement(WFS, "Name");
+            xsw.writeCharacters(property.getKey());
+            xsw.writeEndElement();
+
+            xsw.writeStartElement(WFS, "Value");
+            xsw.writeCharacters(property.getValue());
+            xsw.writeEndElement();
+
+            xsw.writeEndElement();
+        }
+
+        addGeometries(xsw, feature, true);
+        writeFeatureIdFilter(xsw, feature.getId());
+        xsw.writeEndElement(); // close <wfs:Update>
+
+        xsw.writeEndElement();
+        xsw.writeEndDocument();
+        xsw.close();
+    }
+
+    public static void insertFeature(OutputStream out, Feature feature)
+            throws XMLStreamException {
+        XMLStreamWriter xsw = XOF.createXMLStreamWriter(out);
+        writeStartTransaction(xsw, "1.1.0");
+
+        xsw.writeStartElement(WFS, "Insert");
+        xsw.writeStartElement(feature.getLayerName());
+        xsw.writeAttribute("xmlns:" + feature.getNamespace(), feature.getNamespaceURI());
+
+        for (Map.Entry<String, String> property : feature.getProperties().entrySet()) {
+            xsw.writeStartElement(property.getKey());
+            xsw.writeCharacters(property.getValue());
+            xsw.writeEndElement();
+        }
+
+        addGeometries(xsw, feature, false);
+        xsw.writeEndElement();
+        xsw.writeEndElement(); // close <wfs:Insert>
+
+        xsw.writeEndElement();
+        xsw.writeEndDocument();
+        xsw.close();
+    }
+
+    public static void deleteFeature(OutputStream out, Feature feature)
+            throws XMLStreamException {
+        XMLStreamWriter xsw = XOF.createXMLStreamWriter(out);
+        writeStartTransaction(xsw, "1.1.0");
+
+        xsw.writeStartElement(WFS, "Delete");
+        xsw.writeAttribute("typeName", feature.getLayerName());
+        writeFeatureIdFilter(xsw, feature.getId());
+        xsw.writeEndElement(); // close <wfs:Delete>
+
+        xsw.writeEndElement();
+        xsw.writeEndDocument();
+        xsw.close();
+    }
+
+    private static void addGeometries(XMLStreamWriter xsw, Feature feature, boolean isUpdate) throws XMLStreamException {
+        if(!feature.hasGeometry()) {
+            return;
+        }
+
+        if(isUpdate == false) {
```

Comment:
This function is only called from two different places but still it requires a special `isUpdate` flag which totally changes what the function actually does. Also, the insert version (when `isUpdate` is `false`) is only three lines of code. I think it's more readable if we move these three lines here to the `insertFeature` function and wrap them inside a `if (feature.hasGeometry()) {}` block there instead. Then we can remove the `boolean isUpdate` argument and rename the function to something more detailed like `writeGeometryProperty`